source/extensions/filters/http/jwt_authn/filter_config.cc

```diff
@@ -29,6 +33,32 @@ void FilterConfigImpl::init() {
           it.first, Verifier::create(it.second, proto_config_.providers(), *this));
     }
   }
+
+  std::vector<std::string> names;
+  for (const auto& it : proto_config_.requirement_map()) {
+    names.push_back(it.first);
+    name_verifiers_.emplace(it.first,
+                            Verifier::create(it.second, proto_config_.providers(), *this));
+  }
+  // sort is just for unit-test since protobuf map order is not deterministic.
+  std::sort(names.begin(), names.end());
+  all_requirement_names_ = absl::StrJoin(names, ",");
+}
+
+std::pair<const Verifier*, std::string>
+FilterConfigImpl::findPerRouteVerifier(const PerRouteFilterConfig& per_route) const {
+  if (per_route.config().bypass()) {
+    return std::make_pair(nullptr, EMPTY_STRING);
+  }
+
+  const auto& it = name_verifiers_.find(per_route.config().requirement_name());
+  if (it != name_verifiers_.end()) {
+    return std::make_pair(it->second.get(), EMPTY_STRING);
+  }
+
+  return std::make_pair(
```

Comment:
this is used in response_code_details, should we remove the whitespace in it? I remember @alyssawilk mentioned that this might break some log processing that assumes each log segment is separated by whitespace.

Reply:
Good point. Changed filter.cc to replace white space with dash for RcDetails.

Reply:
Hi Wayne, I think we need to change other rcDetails in the JwtAuthn filter(using jwt verification results) or other filters(we have introduced some RcDetails with whitespace recently) to follow this.

Reply:
All of them haven't been released yet for ESPv2.